src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -161,15 +161,14 @@ public AgentSettings LoadSettings()
 
             // Loop getting agent name and pool name
             string poolName = null;
-            int poolId = 0;
-            string agentName = null;
+            AgentSettings agentSettings;
```

Comment:
this should get created at top of the method, so you can set it's value for every config question, like accept eula, server url.